pkgs/by-name/me/mealie/package.nix

```diff
@@ -0,0 +1,123 @@
+{ lib
+, callPackage
+, fetchFromGitHub
+, nixosTests
+, python3
+, writeShellScript
+}:
+
+let
+  version = "1.1.0";
+  src = fetchFromGitHub {
+    owner = "mealie-recipes";
+    repo = "mealie";
+    rev = "v${version}";
+    sha256 = "sha256-k2jMGJ6bdhkXootWl9cEdevRrGfggmHbkJZHZmhNLTM=";
+  };
+
+  frontend = callPackage (import ./mealie-frontend.nix src version) { };
+
+  python = python3.override {
+    packageOverrides = self: super: {
+      pydantic = self.pydantic_1;
+    };
+  };
+
+  # TODO  Add crfpp in runtime dependencies for ingredient quantities NLP
+
+in python.pkgs.buildPythonPackage rec {
+  pname = "mealie";
+  inherit version src;
+  pyproject = true;
+
+  patches = [
+    # Replace hardcoded paths in code with environment variables (meant for inside Docker only)
+    # So we can configure easily where the data is stored on the server
+    ./mealie_statedir.patch
+    ./mealie_init_db.patch
+    ./mealie_logger.patch
+  ];
+
+  nativeBuildInputs = [
+    python.pkgs.poetry-core
+    python.pkgs.pythonRelaxDepsHook
+  ];
+
+  pythonRelaxDeps = true;
+
+  propagatedBuildInputs = with python.pkgs; [
+    aiofiles
+    alembic
+    aniso8601
+    appdirs
+    apprise
+    bcrypt
+    extruct
+    fastapi
+    gunicorn
+    html2text
+    httpx
+    jinja2
+    lxml
+    orjson
+    passlib
+    pillow
+    psycopg2
+    pyhumps
+    pytesseract
+    python-dotenv
+    python-jose
+    python-ldap
+    python-multipart
+    python-slugify
+    pyyaml
+    rapidfuzz
+    recipe-scrapers
+    sqlalchemy
+    tzdata
+    uvicorn
+  ];
+
+  doCheck = true;
+
+  postInstall = let
+    start_script = writeShellScript "start-mealie" ''
+      export STATIC_FILES="${frontend}"
+      ${lib.getExe python.pkgs.gunicorn} "$@" -k uvicorn.workers.UvicornWorker mealie.app:app;
+    '';
```

Comment:
I think `makeWrapper` would make a better fit here. Also, why call it `start-mealie` and not just `mealie`? Executing a file is starting it, so that much is implied.

Reply:
I called it `start-mealie` because it's not the mealie "binary", but just a script whose job is to start mealie.
That being said, I can rename it if you want, not a problem for me

Reply:
If this is they way through which mealie gets started, I think it is fair to call it mealie. Using `makeWrapper`, you could also set the `PYTHONPATH` correctly, so it wouldn't need to be passed in the systemd unit.